pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -29,8 +31,345 @@ import (
+	StoragePoolStatusInvalid StoragePoolState = "Invalid"
+
+	// StoragePoolStatusErrorDuplicate ensures error due to duplicate resource.
+	StoragePoolStatusErrorDuplicate StoragePoolState = "ErrorDuplicate"
+
+	// StoragePoolStatusPending ensures pending task for StoragePool.
+	StoragePoolStatusPending StoragePoolState = "Pending"
+
+	// StoragePoolStatusInit is initial state of StoragePool, before pool creation.
+	StoragePoolStatusInit StoragePoolState = "Init"
+
+	// StoragePoolStatusCreateFailed is state when pool creation failed
+	StoragePoolStatusCreateFailed StoragePoolState = "PoolCreationFailed"
+)
+
+// StoragePoolMode for the storage pool
+type StoragePoolMode string
+
+const (
+	// SharedStoragePoolMode if the pool is to be shared
+	SharedStoragePoolMode StoragePoolMode = "Shared"
+
+	// ExclusiveStoragePoolMode if the pool is exclusive
+	ExclusiveStoragePoolMode StoragePoolMode = "Exclusive"
+)
+
+// DeviceTypeIdentifier for the device type constraint
+type DeviceTypeIdentifier struct {
+	// Model of the device
+	Model string `json:"model,omitempty"`
+
+	// PartNumber of the device
+	PartNumber string `json:"partNumber,omitempty"`
+}
+
+// FilterOptions for device filters
+type FilterOptions struct {
+	Key     string `json:"key,omitempty"`
+	Name    string `json:"name,omitempty"`
+	State   string `json:"state,omitempty"`
+	Exclude string `json:"exclude,omitempty"`
+	Include string `json:"include,omitempty"`
+}
+
+// DataStorageCapabilities for the pool
+type DataStorageCapabilities struct {
+	// AccessModes to be supported by the pool
+	AccessModes []AccessMode `json:"accessModes,omitempty"`
+
+	// ProvisioningPolicies to be supported by the pool
+	ProvisioningPolicies []ProvisioningPolicy `json:"provisioningPolicies,omitempty"`
+
+	// Multipathing scenario to be supported by pool viz, OnlineActive, OnlinePassive
+	Multipathing StoragePoolMultipathingOption `json:"multipathing,omitempty"`
+
+	// Compression to be supported or not, if yes the algorithms
+	// +nullable
+	Compression *[]CompressionAlgorithm `json:"compression,omitempty"`
+
+	// Deduplication to be supported or not
+	// +nullable
+	Deduplication *bool `json:"deduplication,omitempty"`
+}
+
+// DataSecurityCapabilities for the pool
+type DataSecurityCapabilities struct {
+	// MediaEncryption to be supported or not, if yes the algorithms
+	// +nullable
+	MediaEncryption *[]MediaEncryptionAlgorithm `json:"mediaEncryption,omitempty"`
+
+	// DataSanitizationPolicy to be supported, viz Clear, CryptographicErase
+	// +nullable
+	DataSanitizationPolicy *DataSanitizationPolicy `json:"dataSanitizationPolicy,omitempty"`
+}
+
+// IoConnectivityCapabilities for the pool
+type IoConnectivityCapabilities struct {
+	// AccessProtocols to be supported by the pool, viz NVMe, NVMeOverFabrics
+	AccessProtocols []AccessProtocol `json:"accessProtocols,omitempty"`
+}
+
+// IoPerformanceCapabilities for the pool
+type IoPerformanceCapabilities struct {
```

Comment:
[suggest]: Can the name follow the standard naming convention. IMO `Io` should be renamed to `IO`
This should also be done at json tags i.e. `json:averageIOOperationLatencyMicroseconds`

Reply:
The capabilities section has now been moved to a different file, to keep sync between all the schemas. We will make sure these comments are addressed in https://github.com/openebs/device-localpv/pull/47